tools/cli/commands/create.py

```diff
@@ -389,7 +389,13 @@ def ensure_disk_exists(args, gcloud_compute, disk_name, report_errors=False):
             create_disk(args, gcloud_compute, disk_name, report_errors)
         except:
             if args.zone or args.quiet:
-                raise
+                if report_errors:
```

Comment:
That being said, the logic could probably still be flattened to keep it simpler.

I will do that.